index.bs

```diff
@@ -6277,6 +6293,124 @@ The [=remote end steps=] with |command parameters| are:
 
 </div>
 
+#### The emulation.setUserAgentOverride Command ####  {#command-emulation-setUserAgentOverride}
+
+The <dfn export for=commands>emulation.setUserAgentOverride</dfn> command modifies
+User-Agent on the given top-level traversables or user contexts.
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+    <pre class="cddl" data-cddl-module="remote-cddl">
+      emulation.SetUserAgentOverride = (
+        method: "emulation.setUserAgentOverride",
+        params: emulation.SetUserAgentOverrideParameters
+      )
+
+      emulation.SetUserAgentOverrideParameters = {
+        userAgent: text / null,
+        ? contexts: [+browsingContext.BrowsingContext],
+        ? userContexts: [+browser.UserContext],
+      }
+    </pre>
+   </dd>
+   <dt>Result Type</dt>
+   <dd>
+    <code>
+     EmptyResult
+    </code>
+   </dd>
+</dl>
+
+<div algorithm>
+The <dfn export>WebDriver BiDi emulated User-Agent</dfn> steps given
+[=environment settings object=] |environment settings| are:
+
+1. Let |related navigables| be the result of [=get related navigables=] with
+   |environment settings|.
+
+1. For each |navigable| or |related navigables|:
+
+   1. Let |top-level navigable| be |navigable|'s [=navigable/top-level traversable=].
+
+   1. Let |user context| be |top-level navigable|'s [=associated user context=].
+
+   1. For each |session| in [=active BiDi sessions=]:
+
+      1. If |session|'s [=emulated user agent=]'s [=navigable user agent=] contains
+         |top-level navigable|, return |session|'s [=emulated user agent=]'s
+         [=navigable user agent=][|top-level navigable|].
+
+   1. For each |session| in [=active BiDi sessions=]:
+
+      1. If |session|'s [=emulated user agent=]'s [=user context user agent=]
+         contains |user context|, return |session|'s [=emulated user agent=]'s
+         [=user context user agent=][|user context|].
+
+1. For each |session| in [=active BiDi sessions=]:
+
+   1. Let |default emulated user agent| be |session|'s [=emulated user agent=]'s
+      [=default user agent=].
+
+   1. If |default emulated user agent| is not null, return
+      |default emulated user agent|.
+
+1. Return null.
+
+</div>
+
+<div algorithm="remote end steps for emulation.setUserAgentOverride">
+
+The [=remote end steps=] with |command parameters| are:
+
+1. If |command parameters| [=map/contains=] "<code>userContexts</code>"
+   and |command parameters| [=map/contains=] "<code>context</code>",
+   return [=error=] with [=error code=] [=invalid argument=].
+
+1. Let |emulated user agent| be null.
+
+1. If |command parameters| [=map/contains=] "<code>userAgent</code>":
+
+   1. Set |emulated user agent| to |command parameters|["<code>userAgent</code>"].
```

Comment:
userAgent is mandatory so I think we can skip the check (but I can see we used the same pattern in many other emulation commands, where we can probably do the same simplification)
```suggestion
1. Set |emulated user agent| to |command parameters|["<code>userAgent</code>"].
```